### DISCUSSION TITLE
How would you add a job/task queue to PocketBase?

### DISCUSSION THREAD

**Discussion by YugoCode** [pocketbase/pocketbase#2101] [Q&A] [answered]
In many projects you require a job/task queue functionality, where certain requests of users need to be handled in the background, so they get queued and worker processes handle them.

I was wondering what you think about adding this feature to PocketBase (when used as a framework)? Isn't this use case so common that it makes sense to support it out of the box?

If you don't think it makes sense to add it, could you maybe give me/us guidance on how you would implement this functionality to PocketBase (when used as a framework)? For me the key requirement is that, in the end, the whole backend should still be one binary, so I don't want any dependencies to Redis for example.

I found [taskq](https://github.com/vmihailenco/taskq), which supports in-memory as a backend, but it doesn't seem to be maintained anymore. There is also [goblero](https://github.com/didil/goblero), but it doesn't seem to be that popular, so not sure you can use it in production. Do you or anyone else maybe have some recommendations for a suitable library?

Thanks for any feedback! ❤️

**Comment by ganigeorgiev** [comment, marked as answer]
You don't necessary need a specialized tasks/queue library. I personally don't have experience with the libraries you've linked and I'm not sure what to recommend.

A slightly different approach could be to create a simple `queue` collection on your own (eg. with fields `name`, `data` and `status`) and use `time.Tick` or some more advanced cron-like library (eg. [`go-co-op/gocron`](https://github.com/go-co-op/gocron), [`robfig/cron`](https://github.com/robfig/cron), etc.) that for example could iterate on intervals over the `queue` records with "pending"/"unprocessed" status.

Email queue (and maybe a generic job queue) is in my todo, but for now remains a low priority.
